modules/RTC/TraceablePeerConnection.js

```diff
@@ -2182,20 +1988,50 @@ TraceablePeerConnection.prototype.configureVideoSenderEncodings = function(local
     return Promise.allSettled(promises);
 };
 
-TraceablePeerConnection.prototype.setLocalDescription = function(description) {
-    let localDescription = description;
+/**
+ * Set the simulcast stream encoding properties on the RTCRtpSender.
+ *
+ * @param {JitsiLocalTrack} localTrack - the current track in use for which the encodings are to be set.
+ * @returns {Promise<void>} - resolved when done.
+ */
+TraceablePeerConnection.prototype._setEncodings = function(localTrack) {
+    if (localTrack.getType() === MediaType.VIDEO) {
+        return this._updateVideoSenderParameters(() => this._configureSenderEncodings(localTrack));
+    }
 
-    this.trace('setLocalDescription::preTransform', dumpSDP(localDescription));
+    return this._configureSenderEncodings(localTrack);
+};
 
-    // Munge stereo flag and opusMaxAverageBitrate based on config.js
-    localDescription = this._mungeOpus(localDescription);
+/**
+ * Munges the provided description to update the codec order, set the max bitrates (for VP9 K-SVC), set stereo flag
+ * and update the DD Header extensions for AV1.
+ *
+ * @param {RTCSessionDescription} description - The description to be munged.
+ * @returns {RTCSessionDescription} - The munged description.
+ */
+TraceablePeerConnection.prototype._mungeDescription = function(description) {
+    let mungedDescription = description;
+
+    this.trace('RTCSessionDescription::preTransform', dumpSDP(description));
+    mungedDescription = this.tpcUtils.mungeOpus(description);
```

Comment:
Each of these operations re-parses the SDP. Since we are refactoring, doesn't it make sense to parse it in this functiom, and pass the transform object along, so each of the tpcUtils functions can apply transformations, and then after all of them, re-generate the SDP?